pybamm/models/full_battery_models/lithium_ion/electrode_soh.py

```diff
@@ -604,12 +605,10 @@ def _check_esoh_feasible(self, inputs):
         else:
             # address numpy 1.25 deprecation warning: array should have ndim=0
             # before conversion
-            V_lower_bound = float(
-                self.OCV_function.evaluate(inputs={"x": x0_min, "y": y0_max}).item()
-            )
-            V_upper_bound = float(
-                self.OCV_function.evaluate(inputs={"x": x100_max, "y": y100_min}).item()
-            )
+            inputs.update({"x": x0_min, "y": y0_max}, check_already_exists=False)
```

Comment:
this should create a new dictionary instead of updating inputs in place which might cause bugs elsewhere